File: Sources/App/Objects/Accounts/ProfileObject.swift
import Vapor
import Fluent
import struct Foundation.UUID
import struct Foundation.Date

struct ProfileObject: Content {
    var id: String?
    var username: String
    var avatar: String
    var bannerArt: String?
    var bio: String
    var tagline: String
    var works: Int64
    var blogs: Int64
    var followers: Int64
    var following: Int64
    var links: [String]
    var createdAt: Date?

    func toModel() -> Profile {
        let model = Profile()

        model.id = self.id
        model.username = self.username
        model.avatar = self.avatar
        model.bannerArt = self.bannerArt
        model.bio = self.bio
        model.tagline = self.tagline
        model.links = self.links
        model.createdAt = self.createdAt

        return model
    }
}

File: src/models/accounts/profile_object.rs
use serde::{Serialize, Deserialize};
use chrono::{DateTime, Utc};
use super::role::Role;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
#[cfg_attr(feature = "ssr", derive(sqlx::FromRow))]
pub struct ProfileObject {
    pub id: String,
    pub username: String,
    pub avatar: String,
    pub banner_art: Option<String>,
    pub bio: String,
    pub tagline: Option<String>,
    pub links: Vec<String>,
    pub followers: i64,
    pub following: i64,
    pub works: i64,
    pub blogs: i64,
    pub roles: Vec<Role>,
    pub created_at: DateTime<Utc>,
}